shocker.c

```diff
@@ -1,60 +1,161 @@
 #define _GNU_SOURCE
-#include <sys/wait.h> // waitpid
+#include <stdlib.h>
+#include <stdbool.h>
 #include <sched.h> // clone, all the clone Flags
 #include <unistd.h> // execvp, sethostname
 #include <string.h> // strlen
 #include <stdio.h> // puts, printf
 #include <sys/mount.h> // mount
+#include <sys/stat.h> // mkdir
+#include <sys/types.h> // Flags for sys stuff
+#include <sys/mman.h>
+#include <sys/wait.h> // waitpid
 
-#define STACK_SIZE (1024 * 1024)
+#define STACK_SIZE (1024 * 1024)    /* Stack size for cloned child */
```

Comment:
I think 2mb is the typical stack size on linux.

Reply:
also the way I usually define compile time constant is like this:
```
enum { stack_size = 1024 * 1024 };
```
Now this has a number of upsides and downsides and it's definitely not the norm.

The upside is that enum are the closest thing to a compile time constant that you'll get in C as they will usually end up behaving like you want them to. You also avoid all the macro problem such as, did somebody decide to use the simple STACK_SIZE anywhere in their code or will anybody be stupid enough to write `blah STACK_SIZE)` to call the function blah with the stack size?

the downside is that in C, enums are `int` which means signed and, depending on your arch, usually 32 bits. That can byte you if you're not careful.

If you need to control the type then I recommend:
```
static const size_t stack_size = 1024 * 1024;
```
This does not have the same property as a compile time constants but I believe most compilers will pretend like it is because the dumb shit you can do here are pretty much all undefined behaviour I believe.